### PR TITLE
RavenDB-9165: Add OutOfMemoryException handling to recovery tool.

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: tools/Voron.Recovery/Recovery.cs [hunk 1]
```diff
@@ -130,6 +138,16 @@ void PrintRecoveryProgress(long startOffset, byte* mem, byte* eof, DateTime now)
...
                     {
                         se?.Dispose();
+
+                        if (e is OutOfMemoryException && e.InnerException is Win32Exception)
+                        {
+                            writer.WriteLine($"{e.InnerException.Message}. {e.Message}.");
+                            writer.WriteLine();
+                            writer.WriteLine("Journal recovery failed. To continue, please backup your files and run again with --DisableCopyOnWriteMode flag.");
+                            writer.WriteLine("Please note that this is usafe operation and we highly recommend to backup you files.");
+
```
Comment (ayende) [inline tools/Voron.Recovery/Recovery.cs:149]:
```suggestion
e.Data["ReturnCode"] = 0xDEAD;
```

File: tools/Voron.Recovery/CommandLineApp.cs [hunk 2]
```diff
@@ -176,7 +177,19 @@ private static void ConfigureRecoveryCommand()
...
                         }, cts.Token);
-                        recovery.Execute(Console.Out, cts.Token);
+
+                        try
+                        {
+                            recovery.Execute(Console.Out, cts.Token);
+                        }
+                        catch (Exception e)
+                        {
+                            if (e is OutOfMemoryException && e.InnerException is Win32Exception)
```
Comment (ayende) [inline tools/Voron.Recovery/CommandLineApp.cs]:
```suggestion
                            if(e.Data["ReturnCode"] is int returnCode)
                                   return returnCode;
   
```

**Review by ayende — COMMENTED** [review]
Let's avoid code duplication by using the exception itself for this

File: tools/Voron.Recovery/Recovery.cs (hunk 1 above)
Comment (garayx) [reply tools/Voron.Recovery/Recovery.cs:149]: sure

File: tools/Voron.Recovery/CommandLineApp.cs (hunk 2 above)
Comment (garayx) [reply tools/Voron.Recovery/CommandLineApp.cs]: done

**Review by ppekrol — APPROVED** [review]

**Pull request merged**